docs/site/todo-list-tutorial-sqldb.md

```diff
@@ -0,0 +1,133 @@
+---
+lang: en
+title: 'Running on relational database'
+keywords: LoopBack 4.0, LoopBack 4
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/todo-list-tutorial-sqldb.html
+summary:
+  LoopBack 4 TodoList Application Tutorial - Running on Relational Database
+---
+
+If you are running this example using a relational database, here are the extra
+steps. PostgreSQL is used for illustration, it works in a similar way in other
+relational databases.
+
+### Create a new datasource
+
+Run `lb4 datasource`. Specify the datasource name as `psqlds` and select
+`PostgreSQL` as the connector.
+
+```sh
+$ lb4 datasource
+? Datasource name: psqlds
+? Select the connector for psqlds:
+  Redis key-value connector (supported by StrongLoop)
+  MongoDB (supported by StrongLoop)
+  MySQL (supported by StrongLoop)
+❯ PostgreSQL (supported by StrongLoop)
+  Oracle (supported by StrongLoop)
+  Microsoft SQL (supported by StrongLoop)
+  REST services (supported by StrongLoop)
+  ...
+```
+
+### Update the Repository classes using the new datasource
+
+The Repository classes are used to bind the datasource and the models. Since
+we've changed to another datasource, we need to update the repositories which
+are located in the `src\repositories` folder.
+
+In `todo.repository.ts`, inject the new datasource in the constructor as below:
+
+```ts
+constructor(
+    @inject('datasources.psqlds') dataSource: juggler.DataSource,
+    @repository.getter('TodoListRepository')
+    protected todoListRepositoryGetter: Getter<TodoListRepository>,
+  ) {
+```
+
+Make the same changes in `todo-list.repository.ts` and
+`todo-list-image.repository.ts`.
+
+### Create tables using `npm run migrate` command
+
+Run the following commands to create the corresponding tables in the database.
+
+```sh
+$ npm run build
+$ npm run migrate
+```
+
+Now, the database tables are created.
+
+```
+testdb=# \dt
+           List of relations
+ Schema |     Name      | Type  | Owner
+--------+---------------+-------+-------
+ public | todo          | table | root
+ public | todolist      | table | root
+ public | todolistimage | table | root
+```
+
+For details, see the [Database migrations](Database-migrations.md) documentation
+page.
+
+### Alter tables to define the foreign keys
```

Comment:
IIRC, `postgresql` connector already supports foreign key constraints? Can you verify?

Reply:
I ran the `npm run migrate` command and it didn't create the foreign key constraints.  Do I need to do something else?

Reply:
Creating FK requires defining them in the model similar to MySQL. see
https://github.com/strongloop/loopback-connector-postgresql#auto-migrateauto-update-models-with-foreign-keys

Reply:
let me give it a try tomorrow. Thanks @elv1s.

Reply:
@raymondfeng @elv1s , I think I could use some help here.  In `todo.model.ts`, I'm trying to add the property settings that @elv1s mentioned above:
```
@property({
    postgresql: {
      foreignKeys: {
        fk_todo_todoListId: {
          name: 'fk_todo_todoListId',
          entity: 'TodoList',
          entityKey: 'id',
          foreignKey: 'todoListId',
        },
      },
    },
  })
  @belongsTo(() => TodoList)
  todoListId: number;
```
However, it complained that I cannot apply more than one decorator:
`Error: Decorator cannot be applied more than once on Todo.prototype.todoListId` when running `npm run migrate`.  
@elv1s, were you able to get it running?  Am I missing something?  Thanks!

Reply:
I think we need to remove the call to `property()` decorator inside `belongsTo()` so we can add property metadata like you've shown above @dhmlau (see https://github.com/strongloop/loopback-next/issues/2256#issuecomment-470940504).

Reply:
@dhmlau The foreignKeys need to be defined in `@model`.

Issues  #2345, #2256, and PR https://github.com/strongloop/loopback-next/pull/2442 are related to this in that the db field value of `todoListId` cannot be updated when we already have a decorator associated to it.

Multiple decorators are not supported.

```javascript
@model({
  settings: {
    foreignKeys: {
        fk_todo_todoListId: {
          name: 'fk_todo_todoListId',
          entity: 'TodoList',
          entityKey: 'id',
          foreignKey: 'todoListId',
        },
      },
    },
  },
})
```

Reply:
@elv1s @raymondfeng , thanks for the info.  I managed to get it working! :)  Will work on the PR tomorrow.